corehq/apps/analytics/tasks.py

```diff
@@ -241,7 +241,8 @@ def _send_form_to_hubspot(form_id, webuser, hubspot_cookie, meta, extra_fields=N
 
         # these must be submitted after the form to ensure
         # the contact has been created in hubspot
-        update_hubspot_properties_v2(webuser, extra_fields)
+        if extra_fields:
```

Comment:
nit: this check is not necessary since the same check is done inside `update_hubspot_properties_v2`

Reply:
There are other places that call `update_hubspot_properties_v2` so I wanted to both prevent it from being called in this place, but also guard against improper calls elsewhere. Happy to remove that second check if it does not seem necessary.

Reply:
I think it makes sense to do the check inside `update_hubspot_properties_v2`. This is what we're talking about, right?

https://github.com/dimagi/commcare-hq/blob/99af5fb6af2af13e5fa2026f4052223932231fa9/corehq/apps/analytics/tasks.py#L101

since, as you point out, other places call it too and it's better not to have to do that check before every call.

When I first wrote the comment above I was thinking `update_hubspot_properties_v2` was `_track_on_hubspot`. On further inspection I see there is another call before the properties check:

https://github.com/dimagi/commcare-hq/blob/99af5fb6af2af13e5fa2026f4052223932231fa9/corehq/apps/analytics/tasks.py#L255

Is that an expensive call? If so, I can understand the reason why you'd want to check properties first. In that case I'd recommend moving the properties check _before_ that line so other places that call `update_hubspot_properties_v2` will get that optimization too.